weldx/core.py

```diff
@@ -708,3 +711,557 @@ def units(self) -> pint.Unit:
         if isinstance(self._data, xr.DataArray):
             return self._data.data.units
         return self._units
+
+
+# GenericSeries ------------------------------------------------------------------------
+
+# todo
+#  - use xr_check_coords where possible
+#  - extra evaluate function
+#  - __getitem__ : use DataArray.sel
+#  - pandas time types in TimeSeries vs GenericSeries
+#  - asdf base implementation -> xarray.DataArray units checken (U_ statt string)
+#  - add doctests (examples)
+#  - use dimensionality errors
+
+
+class GenericSeries:
+    """Describes a quantity depending on one or more parameters."""
+
+    _allowed_variables: List[str] = None
+    """Allowed variable names"""
+    _required_variables: List[str] = None
+    """Required variable names"""
```

Comment:
as discussed, the default values should be empty containers of their respective types